[torch] Update caffe2/torch/csrc to build under CUDA 13

q10 opened the pull request:
Summary:
Update caffe2/torch/csrc to build under CUDA 13.  

As of CUDA 13, CCCL v3 is the default, and as such, nvToolsExt.h has been moved to  nvtx3/nvtx3.hpp.

This is needed for building FBGEMM_GPU under CUDA 13 (see D86372925)

Test Plan:
```
# Default build
buck build --flagfile fbcode//mode/dev-nosan fbcode//caffe2:_C_impl
buck build --flagfile fbcode//mode/dev-nosan fbcode//caffe2:_C_impl_cuda

# CUDA 13 build
buck build  @//mode/opt -c fbcode.arch=aarch64 -c fbcode.nvcc_arch=b200 -c fbcode.platform010_cuda_version=13.0  fbcode//caffe2:_C_impl
buck build  @//mode/opt -c fbcode.arch=aarch64 -c fbcode.nvcc_arch=b200 -c fbcode.platform010_cuda_version=13.0  fbcode//caffe2:_C_impl_cuda
```

Differential Revision: D86517946




cc @ptrblck @msaroufim @eqy @jerryzh168 @tinglvv

Skylion007 reviewed (approved):
Should we update the OSS NVTX submodule pointer too?

Aidyn-A reviewed (dismissed):
I would recommend you forcing it somewhere in:
https://github.com/pytorch/pytorch/blob/e545ba2d3428b8d793c9869e5abc1eed7c7fbbdb/cmake/Dependencies.cmake#L962-L976

Or even better to completely remove old NVTX API, as NVTX v3 has been around for more than 5 years.

q10 committed:
[torch] Update caffe2/torch/csrc to build under CUDA 13 (#167401)

Summary:

Update caffe2/torch/csrc to build under CUDA 13.  

As of CUDA 13, CCCL v3 is the default, and as such, nvToolsExt.h has been moved to  nvtx3/nvtx3.hpp.

This is needed for building FBGEMM_GPU under CUDA 13 (see D86372925)

Test Plan:
```
# Default build
buck build --flagfile fbcode//mode/dev-nosan fbcode//caffe2:_C_impl
buck build --flagfile fbcode//mode/dev-nosan fbcode//caffe2:_C_impl_cuda

# CUDA 13 build
buck build  @//mode/opt -c fbcode.arch=aarch64 -c fbcode.nvcc_arch=b200 -c fbcode.platform010_cuda_version=13.0  fbcode//caffe2:_C_impl
buck build  @//mode/opt -c fbcode.arch=aarch64 -c fbcode.nvcc_arch=b200 -c fbcode.platform010_cuda_version=13.0  fbcode//caffe2:_C_impl_cuda
```

Differential Revision: D86517946

q10 commented:
> I would recommend you forcing it somewhere in:
> 
> https://github.com/pytorch/pytorch/blob/e545ba2d3428b8d793c9869e5abc1eed7c7fbbdb/cmake/Dependencies.cmake#L962-L976
> 
> Or even better to completely remove old NVTX API, as NVTX v3 has been around for more than 5 years.

I'll do this in a follow-up PR

q10 commented:
@pytorchbot merge

Aidyn-A commented:
I do not mind the follow up, but would definitely prefer a clean land from the beginning.

jeanschmidt commented:
@pytorchbot merge -f "merged internally, restoring shipit"
